bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/common/SafeCallerBuilder.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright (c) 2014,2017 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.core.common;
+
+import java.util.function.Consumer;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+/**
+ * Builder to create a safe-call wrapper for another object.
+ *
+ * @author Simon Kaufmann - initial contribution and API.
+ *
+ * @param <T>
+ */
+@NonNullByDefault
+public interface SafeCallerBuilder<T> {
+
+    public T build();
```

Comment:
Interface methods are `public` by definition, so no need to add the keyword here.
Same for the other methods of this interface.